### ISSUE TITLE
Garbage responses from npm on z/OS

### ISSUE THREAD

**Issue by rich335z** [ibmruntimes/node#83]
I'm running the latest build **node-v0.12.18-os390-s390x**. Receiving garbage responses from npm (and some node) commands. For example:
![image](https://cloud.githubusercontent.com/assets/23561958/22136689/a70b51a8-de9b-11e6-8749-bbda7d935cf0.png)

Piping it to iconv has no effect:
![image](https://cloud.githubusercontent.com/assets/23561958/22136708/c277f98c-de9b-11e6-84be-ce8d0bc52d49.png)

Interesting observation is that an erroneous command produces mostly legible response:
![image](https://cloud.githubusercontent.com/assets/23561958/22136995/ea1215b6-de9d-11e6-95dd-a1cab77856b3.png)

I clean up all orphaned semaphores before issuing commands, so there appears to be no issue with resources:
```
rjack11:/u/rjack11: >ipcs -y

IPC status as of Thu Jan 19 23:10:29 2017
Message Queues:
MNIDS HWIDS CIDSA CPRIV  CKEY GETEX    MAXQB    QMNUM ENOMEM
  500    15   485    14     1     0 2147483647    10000      0
Shared Memory:
MNIDS HWIDS CIDSA CPRIV  CKEY GETEX  TPAGES       SPAGES SEGPR  CPAGES       MAXSEG
  500     4   496     0     4     0  262144        25600   500       0         4096
Semaphores:
MNIDS HWIDS CIDSA CPRIV  CKEY GETEX MNSEMS MNOPS   CSBYTES
  750   183   746     0     4     0   1000    25 100000000
```

```
rjack11:/u/rjack11: >ipcs | wc -l
     31
```

Some basic node commands produce expected response:
```
rjack11:/u/rjack11: >node --help
Usage: node [options] [ -e script | script.js ] [arguments]
       node debug script.js [arguments]

Options:
  -v, --version        print node's version
  -e, --eval script    evaluate script
  -p, --print          evaluate script and print result
  -i, --interactive    always enter the REPL even if stdin
                       does not appear to be a terminal
  --no-deprecation     silence deprecation warnings
  --throw-deprecation  throw an exception anytime a deprecated function is used
  --trace-deprecation  show stack traces on deprecations
  --v8-options         print v8 command line options
  --max-stack-size=val set max v8 stack size (bytes)
  --enable-ssl3        enable ssl3

Environment variables:
NODE_PATH              ':'-separated list of directories
                       prefixed to the module search path.
NODE_MODULE_CONTEXTS   Set to 1 to load modules in their own
                       global contexts.
NODE_DISABLE_COLORS    Set to 1 to disable colors in the REPL

Documentation can be found at http://nodejs.org/
v0.12.18-pre-ea4
rjack11:/u/rjack11: >node --version
v0.12.18
```

However, a `node --v8-options` produces junk after a couple of lines:
![image](https://cloud.githubusercontent.com/assets/23561958/22137353/2c549c58-dea0-11e6-9860-fa78576690db.png)

Running an http hello world works and I get the expected messages in browser, but the console.log output produces garbage:
![image](https://cloud.githubusercontent.com/assets/23561958/22137502/2377813a-dea1-11e6-95d7-9e5d1a90f373.png)

Any help will be appreciated. Let me know what additional doc may be needed.

Thanks,
Rich

**Comment by std3lqi** [comment]
Hi Rich,

You can use `node app 2>&1 | iconv -f ISO8859-1 -t IBM-1047` to see the output.

**Comment by rich335z** [comment]
Thanks @std3lqi but that had no effect. Still getting junk responses

**Comment by zorts** [comment]
This is TOTALLY a stab in the dark, but - what happens if you set _BPXK_AUTOCVT=ON in your environment?

**Comment by jBarz** [comment]
I am wondering if it is possible that the terminal is tagged as ascii.
What does the following show?
ls -T \`tty\`

**Comment by rich335z** [comment]
Thanks @zorts - Actually, I already have _BPXK_AUTOCVT=ON in my profile. However, since you mentioned it, I set it to OFF, and now the responses are readable. So, that's something...

This is a little concerning though because the Git client requires _BPXK_AUTOCVT=ON

**Comment by rich335z** [comment]
@jBarz 
```
rjack11:/u/rjack11: >ls -T `tty`
b binary      T=off /dev/ttyp0000
```

**Comment by zorts** [comment]
@rich335z Yes, Houston, we've got a problem. :-/

I really question the appropriateness of _BPXK_AUTOCVT being controllable in the environment; it really seems to me that the individual APPLICATION ought to know whether it was designed with the whole enhanced ASCII infrastructure in mind or not.

**Comment by jBarz** [comment]
@rich335z : I tagged my terminal as binary and got garbage output from npm.
I removed the binary tag and the output was fixed again. So some of the garbage you see might be due to that.

The garbage output for --v8-options is because we compiled v8 by converting all literals to ascii.
What this means is that when v8 prints out to screen directly, you will see ascii (garbage).
We are in the process of incrementally fixing these error messages in v8 as we notice them.
